plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -2791,10 +2791,19 @@ public function read_meta( &$object ) {
 	 * @return bool
 	 */
 	public function delete_meta( &$object, $meta ) {
-		$delete_meta = $this->data_store_meta->delete_meta( $object, $meta );
-		$this->after_meta_change( $object, $meta );
+		if ( $this->should_backfill_post_record() && isset( $meta->id ) ) {
+			// Let's get the actual meta key before its deleted for backfilling. We cannot delete just by ID because meta IDs are different in HPOS and posts tables.
+			$db_meta = $this->data_store_meta->get_metadata_by_id( $meta->id );
+			if ( $db_meta ) {
+				$meta->key   = $db_meta->meta_key;
+				$meta->value = $db_meta->meta_value;
+			}
+		}
 
-		if ( $object instanceof WC_Abstract_Order && $this->should_backfill_post_record() ) {
+		$delete_meta     = $this->data_store_meta->delete_meta( $object, $meta );
+		$changes_applied = $this->after_meta_change( $object, $meta );
```

Comment:
I'm not sure it's essential, but should we do anything if this returns false? Atm I don't believe we do anything with `$changes_applied`.

Reply:
yeah, this should be included, adding a small commit for this.